File: platforms/android/stato/src/main/java/com/facebook/stato/nativeplugins/table/TablePlugin.java
package com.facebook.stato.nativeplugins.table;

import com.facebook.stato.core.StatoConnection;
import com.facebook.stato.nativeplugins.NativePlugin;
import com.facebook.stato.nativeplugins.RawNativePlugin;

public abstract class TablePlugin implements NativePlugin {

  public abstract TableMetadata getMetadata();

  public void onConnect(TableRowDisplay display) {}

  public void onDisconnect() {};

  @Override
  public final RawNativePlugin asStatoPlugin() {
    return new RawNativePlugin("Table", getTitle()) {

      @Override
      public void onConnect(final StatoConnection connection) throws Exception {
        final TableRowDisplay display = new TableRowDisplayImpl(connection, TablePlugin.this);
        TablePlugin.this.onConnect(display);
      }

      @Override
      public void onDisconnect() throws Exception {
        TablePlugin.this.onDisconnect();
      }

      @Override
      public boolean runInBackground() {
        return false;
      }
    };
    }

}

File: platforms/android/android-stato/src/main/kotlin/org/stato/nativeplugins/table/TablePlugin.kt
package org.stato.nativeplugins.table

import org.stato.core.StatoPluginConnection
import org.stato.nativeplugins.NativePlugin
import org.stato.nativeplugins.RawNativePlugin

abstract class TablePlugin : NativePlugin {

  abstract val metadata: TableMetadata

  fun onConnect(display: TableRowDisplay) {}

  fun onDisconnect() {}

    override fun asStatoPlugin(): RawNativePlugin {
    return object : RawNativePlugin("Table", title) {


      @Throws(Exception::class)
      override fun onConnect(connection: StatoPluginConnection) {
        val display = TableRowDisplayImpl(connection, this@TablePlugin)
        this@TablePlugin.onConnect(display)
      }


      @Throws(Exception::class)
      override fun onDisconnect() {
        this@TablePlugin.onDisconnect()
      }

            override fun runInBackground(): Boolean {
        return false
      }
    }
  }

}
